api/envoy/extensions/transport_sockets/s2a/v3alpha/s2a.proto

```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+package envoy.extensions.transport_sockets.s2a.v3alpha;
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.transport_sockets.s2a.v3alpha";
+option java_outer_classname = "S2aProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Secure Session Agent (S2A)]
+// [#extension: envoy.transport_sockets.s2a]
+
+// Configuration for S2A transport socket. This allows Envoy clients to
+// configure how to offload mTLS handshakes to the S2A service.
+message S2AConfiguration {
+  // The address of the S2A. This can be an IP address or a hostname.
```

Comment:
Does this have a port component optionally?

Reply:
Yes, it can have a port component. Updated the comment to clarify that.

Reply:
Is the address proto appropriate or overkill @htuch?

Reply:
We already discussed that internally @asraa ; I had the same Q as you and the answer is that they want to make use of stuff below TSI layer and can't rely on any of Envoy internal address handling.